### ISSUE TITLE
A sequence-to-sequence approach for document-level relation extraction

### ISSUE THREAD

**Issue by ymgw55** [ymgw55/MLPapers#163]
# Information
### URL
https://arxiv.org/abs/2204.01098
### Authors
[John Giorgi](https://arxiv.org/search/cs?searchtype=author&query=Giorgi%2C+J), [Gary D. Bader](https://arxiv.org/search/cs?searchtype=author&query=Bader%2C+G+D), [Bo Wang](https://arxiv.org/search/cs?searchtype=author&query=Wang%2C+B)
### Published at
ACL 2022 Workshop
### Date of issue
10 Apr 2022
### Related Websites
ACL 2022 Workshop: https://aclweb.org/aclwiki/BioNLP_Workshop

### Code
https://github.com/JohnGiorgi/seq2rel
# Reading Motivation
relation extraction について学ぶ.
# Summary
多くの relation は文を跨いで登場するので, document-level での relation extraction (DocRE) が必要. DocRE では 文内の情報・文を跨ぐ情報 を合わせて entity 間の関係を捉える必要がある. 既存手法に多いのが, pipeline-based な手法で, 入力に entity を用いる. しかし, entity と relation の抽出を同時に学習することで, 効率的にパフォーマンスが向上する. これは学習時にパラメータを共有しているためである. 論文では, seq-to-seq な手法, seq2rel を提案. seq2sel では, entity のアノテーションが必要な pipeline-based な手法に代わり, end-to-end で DocRE (entity 抽出,  指示語の解析, relation extraction) のサブタスクが学習可能. entity hinting と呼ばれるやり方で, 既存の pipeline-based な手法と有名な biomedical 分野のデータセットで性能を比較し, 性能の向上がみられた.  本研究がそのようなデータセットで初めての end-to-end な手法であり, 今後のベンチマークとなる. 
# Background
relation extraction タスクの分かりやすい説明.
![スクリーンショット 2022-04-27 223841](https://user-images.githubusercontent.com/39965001/165545116-87e2ca29-d00f-4329-84b9-8b5ef0cd24de.png)

# Proposed Method
PubMedBERT or BERT を encoder とする. decoder には, single-layer の LSTM を用い, encoder-decoder 間を multi-head attention でつなぐ. 
工夫としては, 
* target の vocabulary を @.DRUG@ などの特殊な token に制限
* decoding の際に, copy mechanism を用いて input 中の token を使えるようにする.
* decoding の際に, 構文ミスがあった場合は確率を低く設定する. (ただしあまり構文ミスはなくほとんど意味がなかった.)
また loss について, 
<img width="385" alt="スクリーンショット 2022-04-28 123905" src="https://user-images.githubusercontent.com/39965001/165671577-fca0e333-9e0f-4055-a017-7583e067e538.png">
permutation sensitive であり, 不必要に decoder の出力の順番を強要する. 
これは, training data 中の頻度が高い token の組み合わせに overfit する恐れがある. 
そのためこの問題を緩和するために, あらかじめ target の relation の順番を文中に登場した順にソートしておき, モデルに decode しやすくする. 

# Experiment
pipeline vs end-to-end には興味がないので略. 

* DocRED での end-to-end なモデルでの比較.
JEREX は文がある relation に当てはまるかどうかの binary relation に焦点を当てていることに注意. （なので完全な end-to-end は seq2rel が初らしい）
![スクリーンショット 2022-04-27 224323](https://user-images.githubusercontent.com/39965001/165668038-b2453fc4-5863-4da9-957c-c1e84f925c99.png)

# Consideration
Future Work として,
* loss に cross-entropy を用いるので,  不必要に decoder の出力の順番を強要してしまう. relation に順番はないので何とかしたい. 過去の研究では 強化学習, unordered-multi-tree decoders, non-autoregressive decoders などで対処しているが, どれも binary & intra-sentence (一文中) の relation extraction にとどまっている.
* 提案モデルでは encoder のみ pre-trained, decoder は randam な初期値から学習. T5, BART などのモデルを用いれば decoder も pre-trained されたパラメータを用いることができる.
